main.go

```diff
@@ -112,34 +118,43 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyhttp.
 }
 
 func (p Proxy) handleRequest(w http.ResponseWriter, r *http.Request) error {
-	token := p.getToken(r)
+	queryToken := p.getTokenFromQueryString(r)
+	queryClaim := p.getClaimFromToken(queryToken)
+	cookieToken := p.getTokenFromCookie(r)
+	cookieClaim := p.getClaimFromToken(cookieToken)
 
-	if token == "" {
-		p.log.Info("no token found, calling management api")
+	if !queryClaim.IsValid && !cookieClaim.IsValid {
+		p.log.Info("no valid token found, calling management api")
 		p.setVar(r, CaddyVarRedirectURL, p.ManagementAPI+p.TokenPath+"?returnTo="+url.QueryEscape(p.Host+r.URL.Path))
 		return nil
 	}
 
-	claim, err := getClaimFromToken(p.Secret, token)
-	if errors.Is(err, jwt.ErrTokenExpired) {
-		p.log.Info("jwt has expired, calling management api")
-		p.setVar(r, CaddyVarRedirectURL, p.ManagementAPI+p.TokenPath+"?returnTo="+url.QueryEscape(p.Host+r.URL.Path))
+	var token string
+	var claim ProxyClaim
+	if queryClaim.IsValid {
+		token = queryToken
+		claim = queryClaim
+	} else if cookieClaim.IsValid {
+		token = cookieToken
+		claim = cookieClaim
+	} else {
+		token = ""
+		claim = ProxyClaim{}
```

Comment:
Is this needed? As this is what the default values are. If want to keep these, I suggest initializing the variables to these values instead.